app/models/course/assessment/question/programming.rb

```diff
@@ -62,15 +62,18 @@ def copy_template_files_to(answer)
   # We restore the original package, but capture the new package into a local for processing by
   # the import job.
   def process_new_package
-    return remove_old_package if attachment.nil?
+    return remove_old_package if attachment_reference.nil?
 
-    new_attachment = attachment
-    restore_attribute!(:attachment)
+    new_attachment_reference = attachment_reference
+    restore_attribute!(:attachment_reference)
 
     execute_after_commit do
-      new_attachment.save!
+      # TODO: Somehow the attachable lost inside the transaction.
+      new_attachment_reference.attachable = self
```

Comment:
Yes that is by design. The attachment will only be associated with the question after the job finishes.

In app/models/course/assessment/question/programming.rb<https://github.com/Coursemology/coursemology2/pull/880#discussion_r54672334>:

>
>      execute_after_commit do
> -      new_attachment.save!
> +      # TODO: Somehow the attachable lost inside the transaction.
> +      new_attachment_reference.attachable = self


Why pass attachments around can fix this ? when you save the attachment the reference also got saved and the it's still null:false.

—
Reply to this email directly or view it on GitHub<https://github.com/Coursemology/coursemology2/pull/880/files#r54672334>.